src/zino/statemodels.py

```diff
@@ -72,6 +75,11 @@ class DeviceState(BaseModel):
     ports: Dict[int, Port] = {}
     alarms: Optional[Dict[AlarmType, int]] = None
     boot_time: Optional[datetime.datetime] = None
+    bgp_style: Optional[BgpStyle] = None
+    bgp_peers: Optional[List[IPAddress]] = []
+    bgp_peer_up_times: Optional[Dict[IPAddress, int]] = dict()
+    bgp_peer_admin_status: Optional[Dict[IPAddress, BgpAdminStatus]] = dict()
+    bgp_peer_oper_states: Optional[Dict[IPAddress, BgpOperState]] = dict()
```

Comment:
I find it hard to resist commenting on the data structures this time around, but I think this note should be moved forward to a future refactor:

We don't live in the flat-dict world of Zino 1 / Tcl any more: This is a bunch of dicts with the same index.  This would probably play better if `BGPPeerSession` was a separate model of attributes concerning a peering session, and then `bgp_peers` would be the single `dict` on `Device` that indexes them.

Reply:
> We don't live in the flat-dict world of Zino 1 / Tcl any more: This is a bunch of dicts with the same index. This would probably play better if `BGPPeerSession` was a separate model of attributes concerning a peering session, and then `bgp_peers` would be the single `dict` on `Device` that indexes them.

Of course, if we push ahead with an alpha release, and change this before the final release, we have to think about whether we want to support migrating from older state data, or just declare that the format isn't going to be stable until a final release.